### PR TITLE
fix(share_plus): Throw error if sharePositionOrigin not in sourceView

### PR DESCRIPTION
**Pull request by tkreuder**
(iPad and maybe Mac): The sharePositionOrigin must be in the coordinate space of sourceView. See https://developer.apple.com/documentation/uikit/uipopoverpresentationcontroller/1622324-sourcerect, else `[LayoutConstraints] Unable to simultaneously satisfy constraints.` appears.

## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by tkreuder** [comment]
resolves #951

**Comment by tkreuder** [comment]
resolves #870

File: packages/share_plus/share_plus/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 4.0.10+2
+
+- iOS: Throw PlatformException when iPad share dialog not appearing (sharePositionOrigin not in sourceView)
+
```
Comment (miquelbeltran) [inline packages/share_plus/share_plus/CHANGELOG.md:4]: Can you make it 4.1.0?

**Review by miquelbeltran — COMMENTED** [review]
Thanks! This will help bring clarity to this issue that seems to affect so many people.

Since `sharePositionOrigin` is not mandatory for iPhones, does it keep working as well for those devices when the param is not provided?

File: packages/share_plus/share_plus/CHANGELOG.md (hunk 1 above)
Comment (tkreuder) [reply packages/share_plus/share_plus/CHANGELOG.md]: Done

**Comment by tkreuder** [comment]
> Thanks! This will help bring clarity to this issue that seems to affect so many people.
> 
> Since `sharePositionOrigin` is not mandatory for iPhones, does it keep working as well for those devices when the param is not provided?

I now made sure, that it raises a `PlatformException`:
- Only if popoverPresentationController is given, which is the case for iPad (not for iPhone) and the arguments sharePositionOrigin value, is not in the parents Rect. This avoids an explicit check, if the device is an iPad and would thus work also on other Apple devices, where this is needed (e.g. in future).
- It raises the `PlatformException` no matter which `Share.share* ` method is used

**Comment by tkreuder** [comment]
image
image

**Comment by tkreuder** [comment]
Can this be merged @miquelbeltran ?

**Comment by miquelbeltran** [comment]
on it

**Comment by tkreuder** [comment]
Good catch and thank you @miquelbeltran  :) 👍

**Comment by miquelbeltran** [comment]
released under 4.3.0, thanks to you!

I ran this on my mac on both iphone and ipad simulators

**Pull request merged**